backend/agentd/session.go

```diff
@@ -390,25 +391,86 @@ func (s *Session) Start() (err error) {
 		}
 	}()
 
-	// Subscribe the session to every configured check subscriptions
-	if err := s.subscribe(s.cfg.Subscriptions); err != nil {
-		return err
-	}
+	lager := logger.WithFields(logrus.Fields{
+		"agent":     s.cfg.AgentName,
+		"namespace": s.cfg.Namespace,
+	})
 
 	// Subscribe the agent to its entity_config topic
 	topic := messaging.EntityConfigTopic(s.cfg.Namespace, s.cfg.AgentName)
-	logger.WithField("topic", topic).Debug("subscribing to topic")
+	lager.WithField("topic", topic).Debug("subscribing to topic")
 	// Get a unique name for the agent, which will be used as the consumer of the
 	// bus, in order to avoid problems with an agent reconnecting before its
 	// session is ended
 	agentName := agentUUID(s.cfg.Namespace, s.cfg.AgentName)
 	subscription, err := s.bus.Subscribe(topic, agentName, s.entityConfig)
 	if err != nil {
-		logger.WithError(err).Error("error starting subscription")
+		lager.WithError(err).Error("error starting subscription")
 		return err
 	}
 	s.entityConfig.subscriptions <- subscription
 
+	// Determine if the entity already exists
+	req := storev2.NewResourceRequest(s.ctx, s.cfg.Namespace, s.cfg.AgentName, (&corev3.EntityConfig{}).StoreName())
+	wrapper, err := s.storev2.Get(req)
+	if err != nil {
+		// We do not want to send an error if the entity config does not exist
+		if _, ok := err.(*store.ErrNotFound); !ok {
+			lager.WithError(err).Error("error querying the entity config")
+			return err
+		}
+		lager.Debug("no entity config found")
+
+		// Indicate to the agent that this entity does not exist
+		meta := corev2.NewObjectMeta(corev3.EntityNotFound, s.cfg.Namespace)
+		watchEvent := &store.WatchEventEntityConfig{
+			Action: store.WatchCreate,
+			Entity: &corev3.EntityConfig{
+				Metadata:    &meta,
+				EntityClass: corev2.EntityAgentClass,
+			},
+		}
+		err = s.bus.Publish(messaging.EntityConfigTopic(s.cfg.Namespace, s.cfg.AgentName), watchEvent)
```

Comment:
Rather than send this sentinel entity, why not simply send back the entity config that the agent sent in the first place?

Reply:
The entity does not send its full config, we can only rely on these few headers 😬 
https://github.com/sensu/sensu-go/blob/ec65b9ffcd0a2b40727dde13f2023c2950102d93/backend/agentd/agentd.go#L237-L242